controllers/operands/csv.go

```diff
@@ -0,0 +1,82 @@
+package operands
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	csvv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/controllers/common"
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/components"
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+)
+
+type csvHandler genericOperand
```

Comment:
I think this is an overkill. Setting a single annotation, not dealing with creating the resource or with the upgrade use-case, I think it's enough to implement the `Operand` interface instead of using the `genericOperand` with a hook. The code will be shorter and more focused.

WDYT?

Reply:
updated, much more clean
although I have some failing operandHandler unit tests failing that I can not understand, can you take a look?